packages/ui/src/views/repo/repo-branch-rules/index.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { useEffect, useReducer } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { NavLink } from 'react-router-dom'
+
+import { Button, ButtonGroup, ControlGroup, Fieldset, Spacer, Text } from '@/components'
+import { IRepoStore, MergeStrategy, SandboxLayout } from '@/views'
+
+import { branchRules } from './components/repo-branch-rules-data'
+import {
+  BranchSettingsRuleBypassListField,
+  BranchSettingsRuleDefaultBranchField,
+  BranchSettingsRuleDescriptionField,
+  BranchSettingsRuleEditPermissionsField,
+  BranchSettingsRuleListField,
+  BranchSettingsRuleNameField,
+  BranchSettingsRuleTargetPatternsField,
+  BranchSettingsRuleToggleField
+} from './components/repo-branch-rules-fields'
+// import { repoBranchSettingsFormSchema } from './components/repo-branch-rules-schema'
+import { branchSettingsReducer } from './reducers/repo-branch-rules-reducer'
+import { BranchRulesActionType, BypassUsersList, RepoBranchSettingsFormFields } from './types'
+
+type BranchSettingsErrors = {
+  principals: string | null
+  statusChecks: string | null
+  addRule: string | null
+  updateRule: string | null
+}
+
+interface RepoBranchSettingsRulesPageProps {
+  isLoading?: boolean
+  handleRuleUpdate: (data: RepoBranchSettingsFormFields) => void
+  apiErrors?: BranchSettingsErrors
+  useRepoRulesStore: () => IRepoStore
+}
+
+export const RepoBranchSettingsRulesPage: React.FC<RepoBranchSettingsRulesPageProps> = ({
+  isLoading,
+  handleRuleUpdate,
+  useRepoRulesStore,
+  apiErrors
+}) => {
+  const { presetRuleData, principals, recentStatusChecks } = useRepoRulesStore()
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    reset,
+    formState: { errors, isValid }
+  } = useForm<RepoBranchSettingsFormFields>({
+    mode: 'onChange',
+    defaultValues: {
+      identifier: '',
+      description: '',
+      pattern: '',
+      patterns: [],
+      state: true,
+      default: false,
+      repo_owners: false,
+      bypass: [],
+      rules: []
+    }
+  })
+
+  const [rules, dispatch] = useReducer(
+    branchSettingsReducer,
+    branchRules.map(rule => ({
+      id: rule.id,
+      checked: false,
+      submenu: [],
+      selectOptions: [],
+      input: ''
+    }))
+  )
+
+  const onSubmit: SubmitHandler<RepoBranchSettingsFormFields> = data => {
+    const formData = { ...data, rules }
+    handleRuleUpdate(formData)
+    reset()
+  }
+
+  useEffect(() => {
+    if (presetRuleData) {
+      reset({
+        identifier: presetRuleData?.identifier || '',
+        description: presetRuleData?.description || '',
+        pattern: '',
+        patterns: presetRuleData?.patterns || [],
+        state: presetRuleData?.state && true,
+        default: presetRuleData?.default || false,
+        repo_owners: presetRuleData?.repo_owners || false,
+        bypass: presetRuleData?.bypass || [],
+        rules: []
+      })
+
+      dispatch({
```

Comment:
same. this should just be a callback from the view, and move the logic into the app layer